src/state/ui/reducer.js

```diff
@@ -3,7 +3,12 @@ import { UI_SELECT_API, UI_SELECT_VERSION } from '../actions';
 import { getDefault } from '../../api';
 import schema from './schema';
 
-const reducer = createReducer( { api: getDefault().name, version: null }, {
+export const defaultState = {
+	api: getDefault().name,
+	version: null,
+};
+
+const reducer = createReducer( defaultState, {
```

Comment:
given that the reducer ultimately governs its own state and structure, I would feel better if _this_ code merged the initial values from cache and from the URL, and then we could leave both of those systems decoupled from each other. this would require making this export something like `makeReducer` that takes both of those and merges them.

```js
export const makeReducer = ( { persistedState = null, urlState = null } ) => {
	const initialState = Object.assign(
		{},
		defaultState,
		persistedState,
	);

	if ( urlState ) {
		initialState.request.bodyParams = urlState.bodyParams;
		initialState.request.method = urlState.method;
		…
	}

	return createReducer( initialState, { … } );
}
```